### PR TITLE
cmd: update push deps example

### LINKED ISSUE #1445: term: stop using Dvcfile
since there's no longer a default stage name in `dvc run` 1.0

This update was started in https://github.com/iterative/dvc.org/pull/1420/commits/435f4103f031e03d7ca194890c4ec59c59430fd3 — needs to be continued in its own PR.

Docs where the term is mentioned:
- [x] push cmd ref. (#1591)
- [x] repro cmd ref. (#1504)
- ~~install/plugins (#1570)~~
- [x] versioning use case tutorial (#1526)
- Double check any other mentions (the one in install/plugins can stay for legacy compatibility)

### PR DESCRIPTION
**Pull request by imhardikj**
Close #1445 by addressing the last pending checkbox:
* [x] push cmd ref.

### CODE REVIEW

File: content/docs/command-reference/push.md [hunk 1]
```diff
@@ -155,37 +155,51 @@ a [pipeline](/doc/command-reference/pipeline) has been setup with these
...
-$ dvc status --cloud
-
-  new:            data/model.p
-  new:            data/matrix-test.p
-  new:            data/matrix-train.p
+$ dvc status -c
+    new:            data/featurize/train.pkl
+    new:            data/featurize/train.pkl
+    new:            data/prepared/train.tsv
+    new:            data/prepared/test.tsv
```
Comment (imhardikj) [inline content/docs/command-reference/push.md:191]: @jorgeorpinel `status`, `pull`, `push` doesn't seem to work for example-get-started repo. I have made up this output myself.

File: content/docs/command-reference/push.md [hunk 2]
```diff
@@ -155,37 +155,51 @@ a [pipeline](/doc/command-reference/pipeline) has been setup with these
...
-data/Posts.xml.zip.dvc
-Posts.xml.dvc
-Posts.tsv.dvc
-Posts-test.tsv.dvc
-matrix-train.p.dvc
-model.p.dvc
-Dvcfile
+$ dvc dag
+         +---------+
+         | prepare |
```
Comment (jorgeorpinel) [inline content/docs/command-reference/push.md:160]: Idk if we want to do this change or just something else completely e.g. `cat dvc.yaml` but seems out of scope for this PR given the title?

File: content/docs/command-reference/push.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md:191]: Should work. What errors are you getting?

But let's not change the example stages. You can make dummy data files and stages based on the existing file and stage names in this example e.g. `Posts.xml`, `data/model.p, etc.

File: content/docs/command-reference/push.md [hunk 3]
```diff
@@ -155,37 +155,51 @@ a [pipeline](/doc/command-reference/pipeline) has been setup with these
...
+    new:            data/prepared/train.tsv
+    new:            data/prepared/test.tsv
 ```
 
 One could do a simple `dvc push` to share all the data, but what if you only
 want to upload part of the data?
 
 ```dvc
-$ dvc push --with-deps matrix-train.p.dvc
+$ dvc push --with-deps featurize
```
Comment (jorgeorpinel) [inline content/docs/command-reference/push.md:198]: Stage name should be something like `test-posts`

File: content/docs/command-reference/push.md [hunk 4]
```diff
@@ -155,37 +155,51 @@ a [pipeline](/doc/command-reference/pipeline) has been setup with these
...
 
 ... Do some work based on the partial update
 
-$ dvc push --with-deps model.p.dvc
+$ dvc push --with-deps evaluate
```
Comment (jorgeorpinel) [inline content/docs/command-reference/push.md:202]: `matrix-train`

File: content/docs/command-reference/push.md (hunk 1 above)
Comment (imhardikj) [reply content/docs/command-reference/push.md]: I can think the problem is with the remote of `example-get-started` repo . This was the error:
`ERROR: unexpected error - Unable to locate credentials`

File: content/docs/command-reference/push.md (hunk 2 above)
Comment (imhardikj) [reply content/docs/command-reference/push.md]: This is actually based on example get started repo. I excluded the `data/data.xml.dvc`, as it was taking to much space. This is similar to [dag example](https://dvc.org/doc/command-reference/dag#examples)

**Comment by imhardikj** [comment]
@jorgeorpinel I have removed `example-get started` repo and instead set up a project for original example using dummy data and files.
I have used `$cat dvc.yaml` here currently but the `dvc dag` visualization of this dummy project can be viewed in this db08e42997734fcc38bf9655eab34f48ce41d655 commit.

File: content/docs/command-reference/push.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md]: Strange. At `dvc status -c`?

File: content/docs/command-reference/push.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md]: What do you mean this is based on Get Started? No, look at the stage names, Posts.xml.dvc, Posts.tsv.dvc, etc.

Anyway... How is this related to removing "Dvcfile" again? The cat dvc.yaml idea wasn't good, now it's even longer. Way too long. Let's just not change this for now and focus on the original PR scope?

File: content/docs/command-reference/push.md (hunk 3 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md]: I see you went for `Posts-tsv`. Not a great stage name. It should be a descriptive action. I left specific suggestions...

File: content/docs/command-reference/push.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md]: Anyway, is this even relevant anymore? 🙂

File: content/docs/command-reference/push.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md]: Yeah just rever this pleas.e It's being taken care of in #1584

Comment (jorgeorpinel) [reply content/docs/command-reference/push.md]: P/a I see it's related because this is where `Dvcfile` gets removed. But yeah, lets just revert thia part since it's being fixed elsewhere already. This way we can merge the rest of this work and move on 🙂

**Comment by shcheklein** [comment]
hey, thanks @imhardikj , but I still see `Dvcfile` in the push.md - I think we should get rid of it completely to close this ticket?

File: content/docs/command-reference/push.md [hunk 5]
```diff
@@ -181,11 +181,11 @@ One could do a simple `dvc push` to share all the data, but what if you only
 want to upload part of the data?
 
 ```dvc
-$ dvc push --with-deps matrix-train.p.dvc
+$ dvc push --with-deps test-posts
```
Comment (shcheklein) [inline content/docs/command-reference/push.md:184]: we need to rewrite the example above, the project structure for this command to make sense?

Comment (shcheklein) [reply content/docs/command-reference/push.md:184]: also, why do change the stage name here, intentional?

File: content/docs/command-reference/push.md [hunk 6]
```diff
@@ -181,11 +181,11 @@ One could do a simple `dvc push` to share all the data, but what if you only
...
 
 ... Do some work based on the partial update
 
-$ dvc push --with-deps model.p.dvc
+$ dvc push --with-deps matrix-train
```
Comment (shcheklein) [inline content/docs/command-reference/push.md:188]: same here - why do we need to change the stage name?

**Comment by jorgeorpinel** [comment]
> thanks @imhardikj , but I still see Dvcfile in the push.md

Sorry that wsa confusing but I set it up that way: someone else was working on that part of the code so I asked Hardik to leave it alone. Now both PRs are merged and `Dvcfile` is gone from the code base 🙂

File: content/docs/command-reference/push.md (hunk 5 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md:184]: > rewrite the example above, the project structure for this command to make sense

Yes! It was also done in the complementary PR. Both are merged, please check in https://dvc.org/doc/command-reference/push#example-with-dependencies

> why do change the stage name here, intentional?

Intentional. I came up with these new names. But now that you mention it `matrix-train` would've been the obvious choice here 🤦

Comment (jorgeorpinel) [reply content/docs/command-reference/push.md:184]: Wait no, that't the name of the next stage. That's shy I came up with `test-posts`.

File: content/docs/command-reference/push.md (hunk 6 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/push.md:188]: I don't think `model` is a good stage name.
